src/core/net/nat64_translator.cpp

```diff
@@ -460,6 +493,91 @@ void Translator::MappingExpirerHandler(Timer &aTimer)
     aTimer.Get<Translator>().mMappingExpirer.Start(kAddressMappingIdleTimeoutMsec);
 }
 
+Error Translator::GetNextAddressMapping(otNat64AddressMappingIterator &aIterator, otNat64AddressMapping &aMapping)
+{
+    Error     err = kErrorNotFound;
+    TimeMilli now = TimerMilli::GetNow();
+
+    for (; aIterator < kAddressMappingPoolSize; aIterator++)
+    {
+        if (mAddressMappingPool.GetEntryAt(aIterator).mExpiry > now)
+        {
+            mAddressMappingPool.GetEntryAt(aIterator).CopyTo(aMapping);
+            aIterator++;
+            ExitNow(err = kErrorNone);
+        }
+    }
```

Comment:
Why not go over `mActiveAddressMappings` entries instead of the pool entries?

Here is my suggestion:
- We can change `otNat64AddressMappingIterator ` type to be a struct containing a `void *`. 
- We can add a public API to `InitIterator()` in which we set it to head of the list.
-  Then `GetNextMapping()` becomes iterating over the linked list.